app/code/Magento/Ui/view/base/web/js/form/element/date.js

```diff
@@ -95,7 +95,7 @@ define([
             timezoneFormat: 'YYYY-MM-DD HH:mm',
 
             listens: {
-                'value': 'onValueChange',
```

Comment:
Hi, @vishal-7037, thank you for collaboration. We can't remove the handler `onValueChange`. 

The `value` property has the responsibility to set and get data from the `registry`. 

The `registry` use as storage to transport data from client to server and conversely, the data from `registry` send to the server and server puts the data to the `registry` when page loading.

All `UiComponents` reads data from `registry` and put it to local `value` property.
You can look it inside `abstract.js`. 

The `shiftedValue` is the only local property that used for UI rendering.

So, in case when we selected some date in `datepicker` and saved it after page reload the `datepicker` component will not render it, because the `value` is exist, but handler for `value` don't run, and date will not be set to the `shiftedValue` property.